src/main/java/seedu/toluist/model/Task.java

```diff
@@ -83,9 +89,11 @@ public boolean equals(Object other) {
                 && this.description.equals(((Task) other).description)) // state check
                 && this.priority.equals(((Task) other).priority)
                 && this.allTags.equals(((Task) other).allTags)
+                && Objects.equals(this.recurringFrequency, ((Task) other).recurringFrequency) // handles null
```

Comment:
what if:
- Same cca on 6pm every thurday starting from this week
- same cca on 6pm every thursday starting from next week

if you mark complete the first one u have 2 identical recurring tasks